core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -491,6 +491,43 @@ public TableMetadata updatePartitionSpec(PartitionSpec newPartitionSpec) {
         currentSnapshotId, snapshots, snapshotLog, addPreviousFile(file, lastUpdatedMillis));
   }
 
+  public TableMetadata updateSortOrder(SortOrder newOrder) {
+    SortOrder.checkCompatibility(newOrder, schema);
+
+    // determine the next order id
+    int newOrderId = INITIAL_SORT_ORDER_ID;
+    for (SortOrder order : sortOrders) {
+      if (order.sameOrder(newOrder)) {
+        newOrderId = order.orderId();
+        break;
+      } else if (newOrderId <= order.orderId()) {
+        newOrderId = order.orderId() + 1;
```

Comment:
Could this cause a problem if I have a table that goes `unsorted` -> `sorted 1,2,3` -> `unsorted` -> `sorted 4,5,6`?  Wouldn't the ids then go `0` -> `1` -> `0` -> `1`, where we have two separate sort orders with the id `1`?

Reply:
I guess if they're only added incrementally with ascending orders the last will always be the largest value (e.g. max(orderIds).